crates/nu-parser/src/parser.rs

```diff
@@ -5268,49 +5241,38 @@ pub fn parse_value(
         SyntaxShape::Range => {
             parse_range(working_set, span).unwrap_or_else(|| garbage(working_set, span))
         }
+        // Check for reserved keyword values
+        SyntaxShape::Nothing | SyntaxShape::Any if bytes == b"null" => {
+            Expression::new(working_set, Expr::Nothing, span, Type::Nothing)
+        }
+        SyntaxShape::Boolean | SyntaxShape::Any if bytes == b"true" => {
+            Expression::new(working_set, Expr::Bool(true), span, Type::Bool)
+        }
+        SyntaxShape::Boolean | SyntaxShape::Any if bytes == b"false" => {
+            Expression::new(working_set, Expr::Bool(false), span, Type::Bool)
+        }
+        SyntaxShape::Filepath
+        | SyntaxShape::Directory
+        | SyntaxShape::GlobPattern
+        | SyntaxShape::String
+            if matches!(bytes, b"true" | b"false" | b"null") =>
```

Comment:
can you add a comment explaining the context for having this here?

Reply:
I have added a TODO note, hopefully it makes sense to you.